### DISCUSSION TITLE
Fetching data from DB

### DISCUSSION THREAD

**Discussion by ssoronid** [tremorlabs/tremor-npm#167] [Q&A]
Which ORM or DB is the best for Tremor? I would like to see a demo or docs connecting them for newbie JS like us.

We should store a const query and then call that const dynamically?

What about an empty state or isLoading state so we can render while the query is being made?

Thanks in advance! I'm in love with this Lib 🙏😬

**Comment by eykrehbein** [comment]
Hi @ssoronid - I think what ORM and DB you use is completely dependent on the type of app you want to build. Traditionally, projects involving dashboards are likely to be built more often with relational databases, I guess. So e.g. MySQL or PostgreSQL would be great options for databases and [Prisma](https://www.prisma.io/) is an awesome ORM for them. If you want to get up and running fast, I can highly recommend [Supabase](https://supabase.com/) as your PostgreSQL hosting provider, since it not only hosts your DB but also user authentication, storage, etc.

However, the options are limitless. MongoDB is a good choice for non-relational databases, TimescaleDB for time-series, etc.

To fetch data, I can highly recommend [React Query](https://tanstack.com/query/v4/docs/adapters/react-query). It takes a bit of reading, but once you understand the basics, it is a very useful and convenient tool.

I hope I was able to help you a little. :)

**Comment by sdairs** [comment]
@ssoronid check out https://www.tinybird.co/ which is a serverless analytics database that publishes your data via HTTP APIs, so you don't need to build the DB->ORM->API bit yourself. Great for powering charts that need timeseries, aggregations, etc.

If you're looking for something transactional (not for analytics) https://www.tigrisdata.com/ is a cool serverless transactional DB.

Both have free tiers to start with.
